internal/util/fscrypt/fscrypt.go

```diff
@@ -0,0 +1,439 @@
+/*
+Copyright 2022 The Ceph-CSI Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fscrypt
+
+/*
+#include <linux/fs.h>
+*/
+import "C"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/user"
+	"path"
+	"time"
+	"unsafe"
+
+	fscryptactions "github.com/google/fscrypt/actions"
+	fscryptcrypto "github.com/google/fscrypt/crypto"
+	fscryptfilesystem "github.com/google/fscrypt/filesystem"
+	fscryptmetadata "github.com/google/fscrypt/metadata"
+	"github.com/pkg/xattr"
+	"golang.org/x/sys/unix"
+
+	"github.com/ceph/ceph-csi/internal/kms"
+	"github.com/ceph/ceph-csi/internal/util"
+	"github.com/ceph/ceph-csi/internal/util/log"
```

Comment:
```suggestion
	"github.com/ceph/ceph-csi/internal/kms"
	"github.com/ceph/ceph-csi/internal/util"
	"github.com/ceph/ceph-csi/internal/util/log"

	fscryptactions "github.com/google/fscrypt/actions"
	fscryptcrypto "github.com/google/fscrypt/crypto"
	fscryptfilesystem "github.com/google/fscrypt/filesystem"
	fscryptmetadata "github.com/google/fscrypt/metadata"
	"github.com/pkg/xattr"
	"golang.org/x/sys/unix"
```

`golang lib - internal - third party `